### PR TITLE
Etna HappyThought project week 7

### PR DESCRIPTION
**Pull request by Caracal23**
## Netlify link
https://dreamy-choux-a67cf0.netlify.app/

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,3 +1,53 @@
...
+  };
+
+  const onNewThought = (newThought) => {
+    setHappyThoughts([newThought, ...happyThoughts]);
+  };
+
+  const onLike = (likedThought) => {
+    setHappyThoughts(
+      happyThoughts.map((happyThought) => {
+        if (happyThought._id === likedThought._id) {
```
Comment (linda-f) [inline src/App.jsx:28]: Great code here. Tried it like this in the beginning too but somehow broke my code. Feeling inspired now to try it next time again this way. 🌟

File: src/App.jsx [hunk 2]
```diff
@@ -1,3 +1,53 @@
...
+        } else {
+          return happyThought;
+        }
+      })
+    );
+    //we need to update the thought that we liked
+  };
+
+  return loading ? (
+    <img src="./Animation.gif" />
```
Comment (linda-f) [inline src/App.jsx:39]: Love that pulsating heart ❤️

File: src/HappyThought.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import LikeButton from "./components/LikeButton";
+import moment from "moment";
+import PropTypes from "prop-types";
+import "animate.css";
+
+const HappyThought = ({ userMessage, onLike }) => {
+  const timeAgo = moment(userMessage.createdAt).fromNow();
```
Comment (linda-f) [inline src/HappyThought.jsx:7]: Nice implementation of moment! ⌛

**Review by linda-f — COMMENTED** [review]
I really like how you did that project and you really went DRY in my opinion. Something I am still trying to get better in. 🙈 
You also implemented some stretch goals very nice.
Maybe something you could have done; since you have such a nice pulsating heart animation is to set a time to the loading stage so also if the loading is very fast, we still get to see the heart a little bit ❤️

**Review by dzc1 — APPROVED** [review]
Etna really cool project! 💪

First, your file structure + logic within your `App.jsx` file is simply great! loving to see the power of props being passed down to childComps and also, the [logic](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExMmI2dnQ3cDgxanAwdHJ1bXdtdDMxbWJldmFoMG9mM2J5MGxhaDNtNCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/JOe7JxOiMg61ogl6fH/giphy.gif).  

Great approach in working with this API and the useEffect hook! 

Keep up the good work!

**Pull request closed without merging**